examples/src/main/java/com/mapbox/navigation/examples/ui/NavigationViewActivity.kt

```diff
@@ -0,0 +1,146 @@
+    }
+
+    override fun onResume() {
+        super.onResume()
+        navigationView.onResume()
+    }
+
+    override fun onStop() {
+        super.onStop()
+        navigationView.onStop()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        navigationView.onPause()
+        stopLocationUpdates()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        navigationView.onDestroy()
+    }
+
+    override fun onBackPressed() {
+        // If the navigation view didn't need to do anything, call super
+        if (!navigationView.onBackPressed()) {
+            super.onBackPressed()
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        navigationView.onSaveInstanceState(outState)
+        super.onSaveInstanceState(outState)
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
+        super.onRestoreInstanceState(savedInstanceState)
+        navigationView.onRestoreInstanceState(savedInstanceState)
+    }
+
+    override fun onNavigationReady(isRunning: Boolean) {
+        if (!isRunning && !::navigationMapboxMap.isInitialized) {
+            ifNonNull(navigationView.retrieveNavigationMapboxMap()) { navMapboxMap ->
+                this.navigationMapboxMap = navMapboxMap
+                this.navigationMapboxMap.updateLocationLayerRenderMode(RenderMode.NORMAL)
+                this.mapboxMap = navMapboxMap.retrieveMap()
+                this.mapboxMap.addOnMapLongClickListener(this@NavigationViewActivity)
+                navigationView.retrieveMapboxNavigation()?.let { this.mapboxNavigation = it }
+                startLocationUpdates()
+            }
+        }
+    }
+
+    override fun onMapLongClick(point: LatLng): Boolean {
+        if (!::origin.isInitialized) {
+            origin = Point.fromLngLat(point.longitude, point.latitude)
+            navigationView.addMarker(origin)
+            return true
+        }
+        if (!::destination.isInitialized) {
+            destination = Point.fromLngLat(point.longitude, point.latitude)
+            navigationView.addMarker(destination)
+            return true
+        }
+        return false
+    }
+
+    private fun startLocationUpdates() {
+        val request = LocationEngineRequest.Builder(1000L)
+            .setFastestInterval(500L)
+            .setPriority(LocationEngineRequest.PRIORITY_HIGH_ACCURACY)
+            .build()
+        try {
+            localLocationEngine.requestLocationUpdates(
+                request,
+                locationEngineCallback,
+                null
+            )
+            localLocationEngine.getLastLocation(locationEngineCallback)
+        } catch (exception: SecurityException) {
+            Timber.e(exception)
+        }
+    }
+
+    private fun stopLocationUpdates() {
+        localLocationEngine.removeLocationUpdates(locationEngineCallback)
+    }
+
+    inner class MyLocationEngineCallback : LocationEngineCallback<LocationEngineResult> {
```

Comment:
@Guardiola31337 - I think this inner class causes the leak because it has the handle of the NavigationViewActivity instance.